fraudserv/service.go

```diff
@@ -138,6 +143,16 @@ func (f *ProofService) Broadcast(ctx context.Context, p fraud.Proof) error {
 	return t.Publish(ctx, bin)
 }
 
+func (f *ProofService) AddVerifier(proofType fraud.ProofType, verifier fraud.Verifier) error {
+	f.verifiersLk.Lock()
+	defer f.verifiersLk.Unlock()
+	if _, ok := f.verifiers[proofType]; ok {
+		return fmt.Errorf("verifier for proof type %s already exist", proofType)
```

Comment:
[non-blocking]

why are we erroring, vs just a no-op if it already exists?

Reply:
either way (no-op or error) is fine. the caller is expected to register a verifier for their proof type right after the ProofService instance is created. in future, if needed we plan to support multiple verifiers per proof type (not needed right now).

Reply:
The error helps if a user thinks multiple validators can be registered, s.t. they actually get an error on the second addition instead of success/no-op. Although, I see where you coming from with this.